### PR TITLE
read all of my shopping lists 

### LINKED ISSUE #2: 2. As a user, I want to read all of my shopping lists and the items in them.
## Summary

The most important feature of our app is the ability to look at their shopping lists to see what's on them, so the user knows what they need to buy. For this to work, we'll have to be able to choose a our shopping lists. The `List` (the active shopping list) and `Home` (where the user will be able to switch shopping lists) components are the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [x] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [x] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` or `SingleList.jsx` to show more information – you will do that **later**.
- The variables you need are already being passed into the `List` and `Home` components. What steps do you need to take in order to render them?
- Additionally, until issues [4](https://github.com/the-collab-lab/tcl-72-smart-shopping-list/issues/4) and [5](https://github.com/the-collab-lab/tcl-72-smart-shopping-list/issues/5) you should only be able to see the shopping list that was manually added in [issue 1](https://github.com/the-collab-lab/tcl-72-smart-shopping-list/issues/1).

### PR DESCRIPTION
**Pull request by ijayhub**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description

- The changes help us to render the name of each shopping list a user subscribed to 
- Helped us render the name of each item in the list
- we learnt more about version control 

## Related Issue

 
- Closes #2 

## Acceptance Criteria

- In Home.jsx, the SingleList component is used to render the name of each shopping list a user is subscribed to
-  In List.jsx, the ListItem component is used to render the name of each item

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.
feature: #2

### Before

- List.jsx
 FireShot Capture 435 - Smart Shopping List - localhost
- Home.jsx
FireShot Capture 433 - Smart Shopping List - localhost

### After

Home.jsx
FireShot Capture 436 - Smart Shopping List - localhost
- List.jsx
Screenshot from 2024-02-06 16-14-02

## Testing Steps / QA Criteria

- pull the branch ho-ij-reading-list and switch to it
- You can see the list from the database both from Home and List components.
- Enter the URL manually to see the path

### CODE REVIEW

File: src/views/Home.jsx [hunk 1]
```diff
@@ -11,6 +12,15 @@ export function Home({ data, setListPath }) {
 				 * TODO: write some JavaScript that renders the `lists` array
 				 * so we can see which lists the user has access to.
 				 */}
+
+				{data?.map((item) => (
```
Comment (khadijahnabil) [inline src/views/Home.jsx:16]: nice use of ?. 🙌

**Review by jmahamed — APPROVED** [review]
Looks good and good use of map method and list item is rendered on the list page

**Review by khadijahnabil — APPROVED** [review]
Great functionality! Works well 🙌

**Comment by llunaplanet** [comment]
Everything working as expected, great work!

**Pull request merged**